src/tests/GC/Features/BackgroundGC/foregroundgc.cs

```diff
@@ -82,8 +86,11 @@ static int Main(string[] args)
 
         static void AllocateTemp()
         {
-            while (!done)
+            while (true)
             {
+                bool check = false;
+                lock(_lock) { check = done; }
+                if (check) { break; }
```

Comment:
```suggestion
                lock(_lock) { if (check) { break; } }
```

nit - no need to write all that out